### DISCUSSION TITLE
Issue when running relation extraction using spacy and LLM

### DISCUSSION THREAD

**Discussion by AGudden** [explosion/spaCy#12814] [Help: Coding & Implementations] [answered]
When trying to use the spacy API for LLN I get following error: OSError: [E053] Could not read meta.json from config.toml

What am I doing wrong?

I created the rel_examples.jsonl file with following content::

{"text": "Laura bought a house in Boston with her husband Mark.", "ents": [{"start_char": 0, "end_char": 5, "label": "PERSON"}, {"start_char": 24, "end_char": 30, "label": "GPE"}, {"start_char": 48, "end_char": 52, "label": "PERSON"}], "relations": [{"dep": 0, "dest": 1, "relation": "LivesIn"}, {"dep": 2, "dest": 1, "relation": "LivesIn"}]}
{"text": "Michael travelled through South America by bike.", "ents": [{"start_char": 0, "end_char": 7, "label": "PERSON"}, {"start_char": 26, "end_char": 39, "label": "LOC"}], "relations": [{"dep": 0, "dest": 1, "relation": "Visits"}]}

I created a config.toml file with content below:

[components.llm.task]
@llm_tasks = "spacy.REL.v1"
labels = ["LivesIn", "Visits"]

[components.llm.task.examples]
@misc = "spacy.FewShotReader.v1"
path = "rel_examples.jsonl"

In my jupyter notebook I have following code:
!python -m pip install spacy-llm
 
s.environ['OPENAI_API_KEY'] = "KEY"
os.environ['OPENAI_API_ORG'] = "KEY"

import spacy
config_path = "config.toml"
nlp = spacy.load(config_path)

When running the code I get the error mentioned above. I would appriciate any help. 

Thanks,
aRNO

**Comment by svlandeg** [comment, marked as answer]
Hi! It'll be easier if you follow the usage example over at https://github.com/explosion/spacy-llm/tree/main/usage_examples/rel_openai.

- Your config file would ideally be in a `.cfg` file and needs to be expanded to cover all subsections that spaCy needs. cf for example https://github.com/explosion/spacy-llm/blob/main/usage_examples/rel_openai/fewshot.cfg
- To load an `nlp` pipeline from the config with `spacy-llm`, you can use our `spacy_llm.util.assemble` utility: `nlp = assemble(config_path)`. cf example code in https://github.com/explosion/spacy-llm/blob/main/usage_examples/rel_openai/run_pipeline.py

Let us know if you run into issues with that!

**Reply by AGudden** [reply]
Hi Sofie, That works beautifuly. Thanks a lot!
